ui/src/admin/actions/index.js

```diff
@@ -306,25 +308,29 @@ export const killQueryAsync = (source, queryID) => (dispatch) => {
   killQueryProxy(source, queryID)
 }
 
-export const deleteRoleAsync = (role, addFlashMessage) => (dispatch) => {
-  // optimistic update
+export const deleteRoleAsync = (role) => async (dispatch) => {
   dispatch(deleteRole(role))
-
-  // delete role on server
-  deleteRoleAJAX(role.links.self, addFlashMessage, role.name)
+  dispatch(publishAutoDismissingNotification('success', 'Role deleted'))
+  try {
+    await deleteRoleAJAX(role.links.self)
```

Comment:
I wonder if the notification should come after this call -- if we report it as deleted, but the AJAX request fails, then we'll publish another notification that we failed to delete it